akka-http/src/main/scala/akka/http/javadsl/server/Route.scala

```diff
@@ -4,11 +4,40 @@
 
 package akka.http.javadsl.server
 
+import akka.stream.javadsl.Flow
+import akka.http.javadsl.model.HttpRequest
+import akka.http.javadsl.model.HttpResponse
+import akka.http.scaladsl
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import akka.http.scaladsl.server.Directives._
+import scala.annotation.varargs
+import akka.NotUsed
+
 /**
- * A marker interface to denote an element that handles a request.
+ * In the Java DSL, a Route can only consist of combinations of the built-in directives. A Route can not be
+ * instantiated directly.
+ *
+ * However, the built-in directives may be combined methods like:
+ * <pre>
+ *
+ * Route myDirective(String test, Supplier<Route> inner) {
+ *     return path("fixed", () ->
+ *         path(test),
+ *             inner
+ *         )
+ *     );
+ * }
  *
- * This is an opaque interface that cannot be implemented manually.
- * Instead, see the predefined routes in [[Directives]] and use the [[Directives.handleWith]]
- * method to create custom routes.
+ * </pre>
+ *
+ * The above example will invoke [inner] whenever the path "fixed/{test}" is matched, where "{test}"
+ * is the actual String that was given as method argument.
  */
-trait Route
\ No newline at end of file
+trait Route {
+  private[http] def toScala: scaladsl.server.Route
```

Comment:
needs `/** INTERNAL API */` comment when has `private[http]` (as it appears as public for Java classes)